Add exposure notifications docs (#683)
OttoWinter/esphomedocs: components/exposure_notifications.rst

BLE Exposure Notification Listener
==================================

.. seo::
    :description: Instructions for setting up exposure notification listeners for ESPHome.
    :image: exposure_notifications.png

The ``exposure_notifications`` component uses the :doc:`/components/esp32_ble_tracker` to discover

> nearby COVID-19 exposure notification bluetooth messages sent by phone's running the

Author: ```suggestion
nearby COVID-19 exposure notification bluetooth messages sent by phones running the
```

oops

`Google/Apple Exposure Notification service <https://www.google.com/covid19/exposurenotifications/>`__.

.. code-block:: yaml

    # Example configuration entry
    esp32_ble_tracker:

    exposure_notifications:
      on_exposure_notification:
        then:
          - lambda: |
              ESP_LOGD("main", "Got notification:");
              ESP_LOGD("main", "  RPI: %s", hexencode(x.rolling_proximity_identifier).c_str());
              ESP_LOGD("main", "  RSSI: %d", x.rssi);

Configuration variables:
------------------------

- **on_exposure_notification** (*Optional*, :ref:`Automation <automation>`): An automation

>   to run when a exposure notification bluetooth message it received.

Author: ```suggestion
  to run when an exposure notification bluetooth message is received.
```

  A variable ``x`` of type ``exposure_notifications::ExposureNotification`` is passed to the automation.
  See the API Reference in the See Also section for more information.

Indicator of device with exposure notifications
-----------------------------------------------

The following configuration can be used as an indicator whether an exposure-notifications
enabled device is nearby. As long as an exposure notification has been received in the last
minute, the indicator will be on.

.. code-block:: yaml

    esp32_ble_tracker:

    switch:
      - platform: gpio
        pin: GPIO22
        id: led

    script:
      - id: start_led
        then:
          - switch.turn_on: led
          - delay: 1min
          - switch.turn_off: led

    exposure_notifications:
      on_exposure_notification:
        then:
          - lambda: |
              ESP_LOGD("main", "Got notification:");
              ESP_LOGD("main", "  RPI: %s", hexencode(x.rolling_proximity_identifier).c_str());
              ESP_LOGD("main", "  RSSI: %d", x.rssi);

          # Stop existing timer so that turn_off doesn't get called
          - script.stop: start_led

>           - script.execute: start_led

Reviewer: it might be worth adding a new verb which does this, in fact, I find it really confusing behavior when two or more instances of a script will run at the same time, I did not find documentation about this "multi instance support" is this something we want to have? why not `script.execute` implicitly stop any running instance and restarts it?

Author: Yeah a verb could be added for that (as well as a general rethink of the names of these actions)

I can't immediately think of a use case for multi-instance scripts, but I do think there are some. Might be because HA also does it this way.

I think the main reason they exist though is recursive scripts. What do you do with a `script.execute` action with itself as the parameter? You can't wait for it, and aborting would not really be a solution.

The existing system isn't *that* bad in terms of stuff it supports, we could really improve the naming here though. But that's for another PR :)
